Assets/Scripts/EyesightSystem/EyeSight.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.PlayerLoop;
+
+[Serializable]
+public class ObjectFoundEvent : UnityEvent<GameObject> {
+}
+
+public class EyeSight : MonoBehaviour {
+    public Vector2 sightDirection;
+    public Vector2 rayOffset;
+
+    [SerializeField] private float rayLength;
+    [SerializeField] private float sightAngle;
+    [SerializeField] private float sightAngleStep;
+    [SerializeField] private LayerMask sightMask;
+    [SerializeField] private LayerMask reportMask;
+    [SerializeField] private ObjectFoundEvent objectFoundEvent;
+
+    void FixedUpdate() {
+        float start = -sightAngle / 2;
+        float end = sightAngle / 2;
+
+        Vector2 startRaycast = new Vector2(transform.position.x + rayOffset.x,
+            transform.position.y + rayOffset.y);
+
+        for (; start <= end; start += sightAngleStep) {
+            Vector2 newDirection = new Vector2(sightDirection.x * Mathf.Cos(start * Mathf.Deg2Rad),
+                sightDirection.x * Mathf.Sin(start * Mathf.Deg2Rad));
+
+            Debug.DrawLine(startRaycast, startRaycast + newDirection * rayLength, Color.red);
+
+            RaycastHit2D hit = Physics2D.Raycast(startRaycast, newDirection, rayLength, sightMask);
```

Comment:
```suggestion
            RaycastHit2D hit = Physics2D.Raycast(startRaycast, newDirection, rayLength, sightMask);
            
#if DEBUG_EYESIGHT_RAYS
            Debug.DrawLine(
                startRaycast, 
                startRaycast + newDirection * (hit ? hit.distance : rayLength), 
                Color.red
            );
#endif
```

This will improve the usefulness of the debug lines, because this will be a more closer representation of what the object sees. (The lines will stop going through walls).

I also **strongly** recommend enclosing ALL the debugging code into a pre-processor condition:

* Debug-drawing consumes resources
* It clutters the screen and you might not need that debug information at the moment

Reply:
I did it, but I am not sure? how to check `DEBUG_EYESIGHT_RAYS`. Can you say, how to check it?

Reply:
To enable a conditional code section, place a `#define` at the very top of the file